src/movement/test/test_thruster_sabertooth.cpp

```diff
@@ -0,0 +1,120 @@
+#include <gtest/gtest.h>
+#include "ros/ros.h"
+#include "robosub/thruster.h"
+#include "utility/serial_testbench.hpp"
+#include "utility/serial.hpp"
+
+typedef struct thruster_info
+{
+  std::string name;
+  uint8_t address;
+  uint8_t port;      // 0 or 1
+}Thruster_info;
+
+Thruster_info *mThruster_info;
+Serial mSerial;
+
+TEST(SerialSubscriber, basicTest)
+{
+    //Variables for testing use.
+    uint8_t address = 0, cmd_byte = 0, speed = 0, checksum = 0;
+    int count = 0;
+    //allocate an array of bytes that will be used to receive serial data from
+    //the serial port. I use the arbitrarily large size of
+    uint8_t serial_data[256] = {0};
+
+    //create an empty thruster message to send to the thruster module
+    robosub::thruster saber_msg;
+
+    //fill out data in the thruster message.
+    saber_msg.data.push_back(1);
+    saber_msg.data.push_back(-1);
+    saber_msg.data.push_back(0.5);
+    saber_msg.data.push_back(0.1);
+    saber_msg.data.push_back(-0.5);
+    saber_msg.data.push_back(-0.1);
+
+    //publish the thruster message
+    pub.publish(saber_msg);
+
+    //after we published the thruster message, the thruster module should have
+    //received the message, and then sent some data down the serial port, here
+    //we read that data and verify it is correct
+    //read one byte from the serial port
+
+    //
+    while(count < 6)
+    {
+        /*
+        if(count < 2)
+        {
+          address = 128;
+        }
+        else if(count < 4)
+        {
+          address = 129;
+        }
+        else
+        {
+          address = 130;
+        } */
+
+        //Gets address
+        address = mThruster_info[count].address
```

Comment:
You are missing a semicolon at the end of this line. Also, where is `mThruster_info` initialized or given data?